crates/ruff_python_formatter/src/module/mod_module.rs

```diff
@@ -37,10 +37,11 @@ impl FormatNodeRule<ModModule> for FormatModModule {
 
     fn fmt_dangling_comments(
         &self,
-        _dangling_comments: &[SourceComment],
+        dangling_comments: &[SourceComment],
         _f: &mut PyFormatter,
     ) -> FormatResult<()> {
-        // Handled as part of `fmt_fields`
+        // Node can't have dangling comments.
+        debug_assert!(dangling_comments.is_empty());
```

Comment:
Not: I think these changes to fmt_dangling_comments could be a separate PR.

Reply:
Yeah, that's fair. I used it to verify that no other node does something similar. But it could certainly have been its own PR. Would you prefer if I split it out?

Reply:
I think it's fine as-is actually -- I assume these debug checks _don't_ pass without these changes.

Reply:
Yeah they don't. That's how I discovered that comprehension attach dangling comments to other nodes in the fist place (because I wanted to add the assertions and then got really confused why some of the nodes suddenly had dangling comments)

Reply:
Totally fine with merging as-is.